### PR TITLE
PubSub node: added tests

### PR DESCRIPTION
**Pull request by irynamatveieva**
## Pull Request description

Added tests for the GCP PubSub node.   

PE PR: https://github.com/thingsboard/thingsboard-pe/pull/2745

## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [x] [Labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#about-labels) that classify your pull request have been added.
- [x] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [x] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [x] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [x] Changes are backward compatible or upgrade script is provided.
- [x] Similar PR is opened for PE version to simplify merge. Crosslinks between PRs added. Required for internal contributors only.

## Back-End feature checklist

- [x] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [x] If new dependency was added: the dependency tree is checked for conflicts.
- [x] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [x] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.
- [x] If new yml property was added: make sure a description is added (above or near the property).

### CODE REVIEW

File: rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/gcp/pubsub/TbPubSubNodeTest.java [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    public void givenErrorOccursDuringInitClient_whenInit_thenThrowsException() throws IOException {
+        willThrow(new RuntimeException("Could not initialize client!")).given(node).initPubSubClient(ctxMock);
+
+        assertThatThrownBy(() -> node.init(ctxMock, new TbNodeConfiguration(JacksonUtil.valueToTree(config))))
+                .isInstanceOf(TbNodeException.class).hasMessage("java.lang.RuntimeException: Could not initialize client!");
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    public void givenForceAckIsTrueAndMessageAttributesPatterns_whenOnMsg_thenEnqueueForTellNext(
```
Comment (ShvaykaD) [inline rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/gcp/pubsub/TbPubSubNodeTest.java:110]: will be nice to add the same test where forceAck is false.

File: rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/gcp/pubsub/TbPubSubNodeTest.java [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+                                "mdAttrName", "mdAttributeName",
+                                "mdAttrValue", "mdAttributeValue"
+                        )), TbMsg.EMPTY_JSON_OBJECT),
+                Arguments.of("$[msgAttrName]", "$[msgAttrValue]", new TbMsgMetaData(),
+                        "{\"msgAttrName\": \"msgAttributeName\", \"msgAttrValue\": \"mdAttributeValue\"}")
+        );
+    }
+
+    @Test
+    public void givenForceAckIsFalseAndErrorOccursOnTheGCP_whenOnMsg_thenTellFailure() {
```
Comment (ShvaykaD) [inline rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/gcp/pubsub/TbPubSubNodeTest.java:156]: will be nice to add the same test where forceAck is true.

File: rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/gcp/pubsub/TbPubSubNodeTest.java [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        assertThatThrownBy(() -> node.init(ctxMock, new TbNodeConfiguration(JacksonUtil.valueToTree(config))))
+                .isInstanceOf(TbNodeException.class).hasMessage("java.lang.RuntimeException: Could not initialize client!");
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    public void givenForceAckIsTrueAndMessageAttributesPatterns_whenOnMsg_thenEnqueueForTellNext(
+            String attributeName, String attributeValue, TbMsgMetaData metaData, String data) {
+        config.setMessageAttributes(Map.of(attributeName, attributeValue));
+        ReflectionTestUtils.setField(node, "forceAck", true);
```
Comment (ShvaykaD) [inline rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/gcp/pubsub/TbPubSubNodeTest.java:113]: please replace reflection with corresponding method stubbing.

File: rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/gcp/pubsub/TbPubSubNodeTest.java [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+                        )), TbMsg.EMPTY_JSON_OBJECT),
+                Arguments.of("$[msgAttrName]", "$[msgAttrValue]", new TbMsgMetaData(),
+                        "{\"msgAttrName\": \"msgAttributeName\", \"msgAttrValue\": \"mdAttributeValue\"}")
+        );
+    }
+
+    @Test
+    public void givenForceAckIsFalseAndErrorOccursOnTheGCP_whenOnMsg_thenTellFailure() {
+        init();
+        ReflectionTestUtils.setField(node, "forceAck", false);
```
Comment (ShvaykaD) [inline rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/gcp/pubsub/TbPubSubNodeTest.java:158]: please replace reflection with corresponding method stubbing.

File: rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/gcp/pubsub/TbPubSubNodeTest.java [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+    public void givenPubSubClientIsNull_whenDestroy_thenShutDownAndAwaitTermination() {
+        ReflectionTestUtils.setField(node, "pubSubClient", null);
+        node.destroy();
+        then(pubSubClientMock).shouldHaveNoInteractions();
+    }
+
+    private void init() {
+        ReflectionTestUtils.setField(node, "config", config);
+        ReflectionTestUtils.setField(node, "pubSubClient", pubSubClientMock);
+    }
```
Comment (ShvaykaD) [inline rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/gcp/pubsub/TbPubSubNodeTest.java:200]: please replace reflection with init method invocation.

**Review by ShvaykaD — CHANGES_REQUESTED** [review]
Please check my comments added above

File: rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/gcp/pubsub/TbPubSubNode.java [hunk 6]
```diff
@@ -129,7 +129,7 @@ private TbMsg processException(TbMsg origMsg, Throwable t) {
         return TbMsg.transformMsgMetadata(origMsg, metaData);
     }
 
-    private Publisher initPubSubClient(TbContext ctx) throws IOException {
+    protected Publisher initPubSubClient(TbContext ctx) throws IOException {
```
Comment (ShvaykaD) [inline rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/gcp/pubsub/TbPubSubNode.java:132]: package-private

Comment (ShvaykaD) [inline rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/gcp/pubsub/TbPubSubNode.java:132]: package-private

**Review by ShvaykaD — CHANGES_REQUESTED** [review]

**Review by ShvaykaD — APPROVED** [review]

**Review by ViacheslavKlimov — APPROVED** [review]

**Pull request merged**